packages/ts/react-grid/src/autoform.tsx

```diff
@@ -0,0 +1,80 @@
+import { type AbstractModel, ValidationError, type DetachedModelConstructor } from '@hilla/form';
+import { Button } from '@hilla/react-components/Button.js';
+import { HorizontalLayout } from '@hilla/react-components/HorizontalLayout.js';
+import { VerticalLayout } from '@hilla/react-components/VerticalLayout.js';
+import { useForm } from '@hilla/react-form';
+import { useEffect, useState, type JSX } from 'react';
+import { AutoFormField } from './autoform-field';
+import type { CrudService } from './crud';
+import { getProperties } from './utils';
+
+type SubmitErrorEvent = {
+  error: unknown;
+};
+type SubmitEvent<TItem> = {
+  item: TItem;
+};
+export type AutoFormProps<TItem> = Readonly<{
+  service: CrudService<TItem>;
+  model: DetachedModelConstructor<AbstractModel<TItem>>;
+  item?: TItem;
+  onSubmitError?({ error }: SubmitErrorEvent): void;
+  onSubmit?({ item }: SubmitEvent<TItem>): void;
+}>;
+
+export function AutoForm<TItem>({ service, model, item, onSubmitError, onSubmit }: AutoFormProps<TItem>): JSX.Element {
+  const form = useForm(model, {
+    onSubmit: async (formItem) => service.update(formItem),
+  });
+  const [formError, setFormError] = useState('');
+  useEffect(() => {
+    form.read(item);
+  }, [item]);
+
+  return (
+    <VerticalLayout theme="padding spacing">
+      {getProperties(model).map((propertyInfo) => (
+        <AutoFormField
+          key={`auto-form-field-${propertyInfo.name}`}
```

Comment:
```suggestion
          key={propertyInfo.name}
```
Nit: The key is never rendered into the DOM, it just needs to be a unique value. As such we can remove any constants from it as those don't make it any more unique.